### PR TITLE
Covered UserMapper and PostMapper with tests

### PR DESCRIPTION
**Pull request by Dmytro-Mykhalko**
…rated IFs that are unreachable.

develop

## GitHub Board

**Issue link**

[UserMapper Test coverage #557](https://github.com/ita-social-projects/dokazovi-be/issues/557)

## Code reviewers
- [ ] @teaFunny 

## Summary of issue

- Analyzed code and uncovered blocks
- Increased coverage of UserMapperImpl up to 93%
- Increased coverage of PostMapperImpl up to 95%
- Couldn't cover some autogenerated IFs that are unreachable
[**Proofs**](https://prnt.sc/sdk6YbUb9Eh1) 

## Summary of change

- updated existing and added extra tests to UserMapperTest
- updated existing and added extra tests to UserMapperTest
- added collectionMappingStrategy to PostMapper, because there was UnsupportedOperationException caused by PostMapperImpl

### CODE REVIEW

File: src/test/java/com/softserveinc/dokazovi/mapper/PostMapperTest.java [hunk 1]
```diff
@@ -118,19 +242,260 @@ void toPostEntity() {
...
+		dtoList.sort(Comparator.comparingInt(DirectionDTOForSavingPost::getId));
+
+		List<DirectionEntity> entityList = new ArrayList<>(entitySet);
+		entityList.sort(Comparator.comparingInt(DirectionEntity::getId));
+
+		for (int i = 0; i < dtoList.size(); i++) {
+			assertEquals(dtoList.get(i).getId(), entityList.get(i).getId());
+		}
+	}
+
```
Comment (V-Kaidash) [inline src/test/java/com/softserveinc/dokazovi/mapper/PostMapperTest.java:500]: Remove empty line

**Review by V-Kaidash — APPROVED** [review]

File: src/test/java/com/softserveinc/dokazovi/mapper/UserMapperTest.java [hunk 2]
```diff
@@ -130,5 +139,126 @@ void toUserDTO() {
...
+		assertNull(userDTO.getMainInstitution());
+		assertNull(userDTO.getSocialNetwork());
+		toUserDtoInstitutionsTest(userDTO, userEntity);
+		toUserDtoDirections(userDTO, userEntity);
+
+		userDTO = mapper.toUserDTO(null);
+		assertNull(userDTO);
+
+	}
+
```
[the remark this answers is not in the document]
Comment (Dmytro-Mykhalko) [reply src/test/java/com/softserveinc/dokazovi/mapper/UserMapperTest.java]: fixed

File: src/test/java/com/softserveinc/dokazovi/mapper/UserMapperTest.java [hunk 3]
```diff
@@ -130,5 +139,126 @@ void toUserDTO() {
...
+		assertNull(userDTO.getBio());
+		assertNull(userDTO.getQualification());
+		assertNull(userDTO.getMainInstitution());
+		assertNull(userDTO.getSocialNetwork());
+		toUserDtoInstitutionsTest(userDTO, userEntity);
+		toUserDtoDirections(userDTO, userEntity);
+
+		userDTO = mapper.toUserDTO(null);
+		assertNull(userDTO);
+
```
[the remark this answers is not in the document]
Comment (Dmytro-Mykhalko) [reply src/test/java/com/softserveinc/dokazovi/mapper/UserMapperTest.java]: fixed

File: src/test/java/com/softserveinc/dokazovi/mapper/PostMapperTest.java (hunk 1 above)
Comment (Dmytro-Mykhalko) [reply src/test/java/com/softserveinc/dokazovi/mapper/PostMapperTest.java]: fixed

**Pull request merged**